scipy/stats/_continuous_distns.py

```diff
@@ -7663,6 +7664,34 @@ def _fitstart(self, data):
 skewcauchy = skewcauchy_gen(name='skewcauchy')
 
 
+# For odd order, the noncentral moments for the skew-normal distribution
+# with location 0 and scale 1 can be expressed as a polynomial in delta,
+# where delta = a/sqrt(1 + a**2) and `a` is the skew-normal shape parameter.
+# This dictionary holds coefficents for that polynomial for orders up to 19.
+# The dict is wrapped in a cached function to reduce the impact of the
+# creation of the dict on import time.
+@lru_cache
+def _get_skewnorm_odd_moments():
```

Comment:
This could be in the class and not use lru but a simple `if` to check if an attribute is filled or not.

Only reason I see to go that route is if we think multiple classes would be created by user and we would want to keep things around in memory.

Reply:
I changed the function to a method of the class.  Initially I kept the use of @lru_cache, but that is overkill--there is only one value, and it is cached the first time it is used.  An `if` could be used, but that is really just an implementation detail for a cached property.  In Python 3.8, there is [`functools.cached_property`](https://docs.python.org/3/library/functools.html#functools.cached_property), which does exactly what we want: the dictionary is constructed the first time it is referenced and becomes an attribute of the instance, so we avoid the cost of creation of the dictionary during import, and subsequent references use the attribute, so there is not even the cost of an `if` statement.

Reply:
Fine by me